File: app/src/main/java/com/ryanrvldo/moviecatalogue/data/source/local/database/TvShowDao.java
package com.ryanrvldo.moviecatalogue.data.source.local.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.ryanrvldo.moviecatalogue.data.source.model.TvShow;

import java.util.List;

@Dao
public interface TvShowDao {

    @Insert
    void insert(TvShow tvShow);

    @Delete
    void delete(TvShow tvShow);

    @Query("SELECT * FROM favorite_tv")
    LiveData<List<TvShow>> getAllFavoriteTv();

    @Query("SELECT * FROM favorite_tv WHERE id=:id")
    TvShow selectById(int id);
}

File: app/src/main/java/com/ryanrvldo/moviecatalogue/data/local/database/TvShowDao.kt
package com.ryanrvldo.moviecatalogue.data.local.database

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query
import com.ryanrvldo.moviecatalogue.data.model.TvShow

@Dao
interface TvShowDao : BaseDao<TvShow> {

    @Query("SELECT * FROM favorite_tv")
    fun getFavoriteTvShows(): LiveData<List<TvShow>>

    @Query("SELECT EXISTS(SELECT * FROM favorite_tv WHERE id=:id)")
    suspend fun isTvShowExists(id: Int): Boolean

}